formats/czi-extract-plugin/src/polus/plugins/formats/czi_extract/czi.py

```diff
@@ -0,0 +1,163 @@
+
+    When CZI files are loaded by BioFormats, it will generally try to mosaic
+    images together by stage position if the image was captured with the
+    intention of mosaicing images together. At the time this function was
+    written, there was no clear way of extracting individual FOVs so this
+    algorithm was created.
+
+    Every field of view in each z-slice, channel, and timepoint contained in a
+    CZI file is saved as an individual image.
+
+    Args:
+        file_path : Path to CZI file
+        out_path : Path to output directory
+    """
+    logger.info("Starting extraction from " + str(file_path.name) + "...")
+
+    base_name = Path(file_path.name).stem
+
+    # Load files without mosaicing
+    czi = czifile.CziFile(file_path, detectmosaic=False)
+    subblocks = [
+        s for s in czi.filtered_subblock_directory if s.mosaic_index is not None
+    ]
+
+    ind: dict = {"X": [], "Y": [], "Z": [], "C": [], "T": [], "Row": [], "Col": []}
+
+    # Get the indices of each FOV
+    for s in subblocks:
+        scene = [dim.start for dim in s.dimension_entries if dim.dimension == "S"]
+        if scene is not None and scene[0] != 0:
+            continue
+
+        for dim in s.dimension_entries:
+            if dim.dimension == "X":
+                ind["X"].append(dim.start)
+            elif dim.dimension == "Y":
+                ind["Y"].append(dim.start)
+            elif dim.dimension == "Z":
+                ind["Z"].append(dim.start)
+            elif dim.dimension == "C":
+                ind["C"].append(dim.start)
+            elif dim.dimension == "T":
+                ind["T"].append(dim.start)
+
+    row_conv = dict(
+        zip(
+            np.unique(np.sort(ind["Y"])),
+            range(0, len(np.unique(ind["Y"]))),
+        ),
+    )
+    col_conv = dict(
+        zip(
+            np.unique(np.sort(ind["X"])),
+            range(0, len(np.unique(ind["X"]))),
+        ),
+    )
+
+    ind["Row"] = [row_conv[y] for y in ind["Y"]]
+    ind["Col"] = [col_conv[x] for x in ind["X"]]
+
+    with BioReader(file_path) as br:
+        metadata = br.metadata
+        chan_names = br.cnames
+
+    for s, i in zip(subblocks, range(0, len(subblocks))):
+        z = None if len(ind["Z"]) == 0 else ind["Z"][i]
+        c = None if len(ind["C"]) == 0 else ind["C"][i]
+        t = None if len(ind["T"]) == 0 else ind["T"][i]
+
+        out_file_path = out_path.joinpath(
+            _get_image_name(
+                base_name,
+                row=ind["Row"][i],
+                col=ind["Col"][i],
+                z=z,
+                c=c,
+                t=t,
+            ),
+        )
+
+        dims = [
+            _get_image_dim(s, "Y"),
+            _get_image_dim(s, "X"),
+            _get_image_dim(s, "Z"),
+            _get_image_dim(s, "C"),
+            _get_image_dim(s, "T"),
+        ]
+
+        data = s.data_segment().data().reshape(dims)
+
+        write_thread(out_file_path, data, metadata, chan_names[c])
```

Comment:
You've done all the ground work for making this multithreaded. Please actually use preadator here.